### PR TITLE
[eCom Tutorial] Refactor to function components and use `loadStripe` helper.

### PR DESCRIPTION
**Pull request by thorsten-stripe**
## Description

I've updated the [demo site](https://github.com/thorsten-stripe/ecommerce-gatsby-tutorial/pull/20) to use function components, and use the new `stripe-js` module:

### Documentation

- Refactor from class components to function components.
- Remove `gatsby-stripe-plugin` see deprecation note: https://www.npmjs.com/package/gatsby-plugin-stripe
- Use loadStripe helper instead: https://github.com/stripe/stripe-js#loadstripe

### CODE REVIEW

**Comment by laurieontech** [comment]
Thanks for the PR! We'll want to verify the technical instructions, so I've added this to our list for doing so.

File: docs/tutorial/ecommerce-tutorial/index.md [hunk 1]
```diff
@@ -433,101 +372,86 @@ const formatPrice = (amount, currency) => {
...
 ```jsx:title=src/components/Products/Skus.js
-import React, { Component } from "react"
+import React from "react"
 import { graphql, StaticQuery } from "gatsby"
-import SkuCard from "./SkuCard" // highlight-line
+import SkuCard from "./SkuCard"
+import { loadStripe } from "@stripe/stripe-js"
 
-const containerStyles = {
+const conatinerStyles = {
```
Comment (AishaBlake) [inline docs/tutorial/ecommerce-tutorial/index.md]:
```suggestion
const containerStyles = {
```

File: docs/tutorial/ecommerce-tutorial/index.md [hunk 2]
```diff
@@ -433,101 +372,86 @@ const formatPrice = (amount, currency) => {
...
-            ))}
-          </div>
-        )}
-      />
-    )
-  }
+        }
+      `}
+      render={({ skus }) => (
+        <div style={conatinerStyles}>
```
Comment (AishaBlake) [inline docs/tutorial/ecommerce-tutorial/index.md]:
```suggestion
        <div style={containerStyles}>
```

File: docs/tutorial/ecommerce-tutorial/index.md [hunk 3]
```diff
@@ -36,68 +35,27 @@ gatsby new ecommerce-gatsby-tutorial
...
+```js
+import { loadStripe } from "@stripe/stripe-js"
 
-```html
-<script src="https://js.stripe.com/v3/"></script>
+const stripe = await loadStripe("pk_test_TYooMQauvdEDq54NiTphI7jx")
 ```
 
-to the end of the `<body>` tag across all of your pages. This helps facilitate Stripe's [fraud detection](https://stripe.com/docs/stripe-js/reference#including-stripejs).
+Stripe.js is loaded as a side effect of the `import '@stripe/stripe-js';` statement. To best leverage Stripe’s advanced fraud functionality, ensure that Stripe.js is loaded on every page of your customer's checkout journey, not just your checkout page. This allows Stripe to detect anomalous behavior that may be indicative of fraud as customers browse your website.
```
[the remark this answers is not in the document]
Comment (thorsten-stripe) [reply docs/tutorial/ecommerce-tutorial/index.md]: Good call, thanks, will update to Stripe.js throughout 👍

File: docs/tutorial/ecommerce-tutorial/index.md [hunk 4]
```diff
@@ -12,20 +15,16 @@ Benefits of using Gatsby for e-commerce sites include the following:
 - Blazing fast performance when your pages are converted from React into static files
 - Easy to host
```
[the remark this answers is not in the document]
Comment (thorsten-stripe) [reply docs/tutorial/ecommerce-tutorial/index.md]: Good catch, thanks, will update throughout. Yes, these talking points are more about why use Gatsby for eCommerce use cases. I'll add a callout that with the client-only Checkout no server component is needed.

**Comment by thorsten-stripe** [comment]
@AishaBlake thanks so much for the review and suggestions :) I've incorporated them and added the sample code to the examples directory. PTAL

**Comment by thorsten-stripe** [comment]
Thank you, appreciate your help with this! :)

**Comment by AishaBlake** [comment]
@thorsten-stripe would you mind verifying [my interpretation of "checkout journey"](https://github.com/gatsbyjs/gatsby/issues/21528#issuecomment-597166407)? This sentence could use some clarification either way.

**Comment by thorsten-stripe** [comment]
@AishaBlake yes, your explanation is spot on. I might include the product catalogue overview and cart pages (if you have any of those).

Also, it looks like we need to updatehttps://www.gatsbyjs.org/docs/processing-payments-with-stripe/. I didn't even know that existed 🙈

**Pull request merged**